feat: add contact notes to case contact card

elasticspoon opened the pull request:
### What github issue is this PR for, if any?
Resolves #5784

### What changed, and _why_?
Created a component to enable text truncation behavior as well as showing the full text. 

Shows notes and answers to contact topics on a per case contact basis.

### How is this **tested**? (please write tests!) 💖💪
component and view specs


### Screenshots please :)
![image](https://github.com/rubyforgood/casa/assets/14540596/4d8e4282-0a37-42bf-bff8-716bef32f230)

### Notes

Truncation is done with CSS rather than server side. I think this is a better approach because it makes the truncation dynamic based on screen size. In this particular use case, I specifically wanted the component to be at most one line in size. 

Given that, maybe it could be named a bit better? I am open to suggestions.

elasticspoon committed:
feat: add contact notes to case contact card

Created a component to enable text truncation behavior as well as
showing the full text. This is done with CSS rather than server side.

Shows notes and answers to contact topics on a per case contact basis.

elasticspoon committed:
fix: removes outdated tests

Testing for truncation is done in component specs, these tests can be
removed.

HTML should be escaped if it is submitted in notes. It is unclear to me
why we are allowing HTML to get parsed and displayed.

schoork commented on app/views/case_contacts/_case_contact.html.erb:
```diff
@@ -21,19 +21,25 @@
+                <% contact.contact_topic_answers.reject { _1.value.blank? }.each do |answer| %>
+                  <li>
+                    <%= render(TruncatedTextComponent.new) do %>
+                      <span class="text-bold d-inline"><%= answer.contact_topic.question %>:</span>
+                      <%= answer.value %>
```
Why not move these into the component's template?

schoork commented on app/components/truncated_text_component.rb:
```diff
@@ -0,0 +1,18 @@
+    @render_check = render_check
+  end
+
+  def text_content
+    return content if content.present?
+
+    simple_format(@text)
+  end
+
+  def render?
+    @render_check && (@text.present? || content.present?)
+  end
```
(Lines 9-17) Unless you're seeing a future use of these, I'd consider removing them, since you're not using them now.

schoork reviewed (approved):
A few comments, but overall looks great!

elasticspoon committed:
refactor: make truncation component less generic

My initial idea was to make a generic truncation component that could be
reused in many places. This led to a more complex interface for using
the component.

At this point it is only used in one place so it makes the most sense to
tailor it to that use case.
